src/ui/game.rs

```diff
@@ -469,31 +470,27 @@ pub fn update_player1_ui(
         Query<&mut Style, With<LevelUI>>,
         Query<&mut Style, (With<ShieldsUI>, With<Player1UI>)>,
     )>,
-    player_query: Query<&PlayerComponent>,
+    player_query: Query<(&HealthComponent, &PlayerComponent)>,
```

Comment:
WTF even is this function? 
```rust
#[allow(clippy::type_complexity)]
/// Update ui to current data from game
pub fn update_player2_ui(
    mut player2_ui_queries: ParamSet<(
        Query<&mut Style, (With<HealthUI>, With<Player2UI>)>,
        Query<&mut BackgroundColor, (With<ArmorUI>, With<Player2UI>)>,
        Query<(&mut BackgroundColor, &mut Transform, &mut PowerGlowUI), With<Player2UI>>,
        Query<&mut Style, (With<AbilityUI>, With<Player2UI>)>,
        Query<&mut Visibility, (With<AbilityChargingUI>, With<Player2UI>)>,
        Query<&mut Visibility, (With<AbilityReadyUI>, With<Player2UI>)>,
        Query<&mut Style, (With<ShieldsUI>, With<Player2UI>)>,
    )>,
    player_query: Query<(&HealthComponent, &PlayerComponent)>,
    time: Res<Time>,
) {
```
Some of these "behavior" functions might be doing to much.

Reply:
This was getting 2 player ui ready before an event. We'll tidy this up when we get to the inevitable UI overhaul.